### ISSUE TITLE
update escalar to be able to handle elasticsearch 5

### ISSUE THREAD

**Issue by raffy4284** [Workday/escalar#15]
Let's go update escalar to be able to use es5!
Now the question is...do we want to package this as a separate jar? or...right a separate class that extends from the Generic classes as specified in this pr: https://github.com/Workday/escalar/pull/7/files and essentially define what es5 client should be using these generic classes?
I vote for the former!

**Comment by sulimmesh** [comment]
Ideally, we'd release separate jars. Our `Generic` classes were all written when we wanted a short-term solution for ES 5 support. ES 1.7 is end of life as of January of this year, so it's not worth it to focus development effort to support it long-term. The only real benefit I can see of genericized escalar is that it makes it easier to adapt to breaking changes in either Jest or the API. But is this really necessary? I'm not sure.

Using the former approach, are you suggesting introducing a branching policy for releasing versioned jars?

**Comment by raffy4284** [comment]
well, I was suggesting the versioned jars, and maybe with the current master putting that to a separate branch ( say ES1.7 branch )
From after that point, then master will now be a ES5;
and in sonatype (? I'm assuming this is where we get our jars ), once we have changed it to use es5, then create a jar that's essentially escalar-5.0 ( escalar that uses elasticsearch 5) 

And you're suggesting of letting go of genericized and instead just stick to just regular AliasInfo, EsError, etc and update these classes that would fit Es5?
Well, given that we're not making escalar backwards compatible ( and just saving es1.7 into a separate jar ), I think i'm ok with this

**Comment by sulimmesh** [comment]
Yes, since we're essentially bringing escalar 1.7 to end of life, we don't need to maintain the approach meant to support it. We'll have the 1.7 jar and can keep it in a branch if we really needed to downgrade for some incredible reason.

Great, I like this approach!
